app/src/main/java/com/expidev/gcmapp/support/v4/fragment/SettingsFragment.java

```diff
@@ -169,6 +187,16 @@ private void updateMccPrefSummary() {
         }
     }
 
+    private class MinistryChangeListener implements Preference.OnPreferenceChangeListener {
+        @Override
+        public boolean onPreferenceChange(Preference preference, Object newValue) {
+            // update summary as necessary
+            updateMinistryPrefSummary();
```

Comment:
for the ministry list preference, it is handled completely by the stock preference support. I just needed to update the format aware summary string.

hmm, I just realized this may be triggered before the value is saved, checking now.